flow/layers/physical_shape_layer.cc

```diff
@@ -43,6 +43,8 @@ void PhysicalShapeLayer::set_path(const SkPath& path) {
 
 void PhysicalShapeLayer::Preroll(PrerollContext* context,
                                  const SkMatrix& matrix) {
+  context->total_elevation += elevation_;
+  total_elevation_ = context->total_elevation;
```

Comment:
You may need to substract context->total_elevation in the end of `Preroll` so its siblings get the right `total_elevation`? I'm thinking that a flow layer unit test could be useful to guard changes like this :)

Reply:
Ah, I just realized that [`ContainerLayer::PrerollChildren`](https://github.com/flutter/engine/blob/master/flow/layers/container_layer.cc#L30) copies the context for each child so it's not needed to subtract the `elevation_` here in the end. That copy seems to be unnecessary as we no longer put `child_paint_bounds` inside the context. But anyway, it makes this change correct.

Also don't worry too much about the unit test. I just realized that there are very few unit tests for the layer so it might take you some time to figure out all the details. Feel free to land this and I'll add the unit test later.